refactor: improve README structure and clarity (#278)
iotaledger/notarization: README.md

Reviewer: IMO this content has already been documented.

### I want a single notarized record

Choose this path for one on-chain proof object, such as a document hash, immutable record, or dynamic latest-state record.

- [Rust client](./notarization-rs)
- [Move contracts](./notarization-move)
- [Wasm bindings](./bindings/wasm/notarization_wasm)
- [Examples](./bindings/wasm/notarization_wasm/examples/README.md)

### I want an audit trail

Choose this path for structured record histories with permissions, capabilities, tagging, and write or delete controls.

> 
> - [Rust client](./audit-trail-rs)
> - [Move contracts](./audit-trail-move)
> - [Wasm bindings](./bindings/wasm/audit_trail_wasm)
> - [Examples](./bindings/wasm/audit_trail_wasm/examples/README.md)
> 
> ### I want the on-chain contracts
> 
> - [Notarization contracts](./notarization-move)
> - [Audit trail contracts](./audit-trail-move)
> 
> ### I want to build an application
> 
> - [Rust client for notarized records](./notarization-rs)
> - [Rust client for audit trails](./audit-trail-rs)
> - [Wasm bindings for notarized records](./bindings/wasm/notarization_wasm)
> - [Wasm bindings for audit trails](./bindings/wasm/audit_trail_wasm)

Reviewer: ```suggestion
### Where To Start

Have a look into [Single Notarization Resources](#single-notarization) for one on-chain proof object, such as a document hash, immutable record, or dynamic latest-state record.

Have a look into [Audit Trails Resources](#audit-trails) for structured record histories with permissions, capabilities, tagging, and write or delete controls.

If  you need to integrate the Notarization on-chain contracts into your own Move package:

- [Notarization contracts](./notarization-move)
- [Audit trail contracts](./audit-trail-move)

If you want to build a client application:

- [Rust client for notarized records](./notarization-rs)
- [Rust client for audit trails](./audit-trail-rs)
- [Wasm bindings for notarized records](./bindings/wasm/notarization_wasm)
- [Wasm bindings for audit trails](./bindings/wasm/audit_trail_wasm)
```

## Documentation and Resources

### Single Notarization

- [Rust package README](./notarization-rs/README.md)

> - [Rust API documentation](https://iotaledger.github.io/notarization/notarization/index.html)

Reviewer: ```suggestion
- [Rust API documentation](https://iotaledger.github.io/notarization/notarization/index.html)
- [Rust Examples](./examples/README.md)
```

- [Move package README](./notarization-move/README.md)
- [Wasm package README](./bindings/wasm/notarization_wasm/README.md)

> - [Examples](./bindings/wasm/notarization_wasm/examples/README.md)

Reviewer: ```suggestion
- [Wasm Examples](./bindings/wasm/notarization_wasm/examples/README.md)
```

- [Docs portal](https://docs.iota.org/developer/iota-notarization)

### Audit Trails

- [Rust package README](./audit-trail-rs/README.md)

> - [Rust API documentation](https://iotaledger.github.io/notarization/audit_trails/index.html)

Reviewer: ```suggestion
- [Rust API documentation](https://iotaledger.github.io/notarization/audit_trails/index.html)
- [Rust Examples](./examples/audit-trail/README.md)
```

- [Move package README](./audit-trail-move/README.md)
- [Wasm package README](./bindings/wasm/audit_trail_wasm/README.md)

> - [Examples](./bindings/wasm/audit_trail_wasm/examples/README.md)

Reviewer: ```suggestion
- [Wasm Examples](./bindings/wasm/audit_trail_wasm/examples/README.md)
```
